NationStatesAPIBot/NationStatesAPIBot/Manager/PermissionManager.cs

```diff
@@ -38,7 +38,11 @@ public Task RemoveRoleAsync(Role role, BotDbContext dbContext)
 
         public async Task RevokePermissionAsync(string discordUserId, Permission permission, BotDbContext dbContext)
         {
-            throw new NotImplementedException();
+            var user = dbContext.Users.Single(u => u.DiscordUserId == discordUserId);
```

Comment:
Single throws an exception when the specified user couldn't be found. 
But you don't handle such exceptions.
Tbh me either lol.

Either handle such exceptions (or at least log them) or use FirstOrDefault/SingleOrDefault (does that exist?) to get null and handle such null objects.

And use async methods where they are possible. Don't know if SingleOrDefaultAsync exist but FirstOrDefaultAsync do.